crates/core/src/db/queries/collections.rs

```diff
@@ -67,3 +67,61 @@ fn make_by_volume_query_string(order_direction: OrderDirection) -> String {
         order_direction = order_direction
     )
 }
+
+/// Query collections ordered by market cap
+///
+/// # Errors
+/// returns an error when the underlying queries throw an error
+pub fn by_market_cap(
+    conn: &Connection,
+    addresses: impl ToSql<Nullable<Array<Text>>, Pg>,
+    order_direction: OrderDirection,
+    limit: impl ToSql<Integer, Pg>,
+    offset: impl ToSql<Integer, Pg>,
+) -> Result<Vec<Nft>> {
+    diesel::sql_query(make_by_market_cap_query_string(order_direction))
+        .bind(addresses)
+        .bind(limit)
+        .bind(offset)
+        .load(conn)
+        .context("Failed to load collections by volume")
+}
+
+fn make_by_market_cap_query_string(order_direction: OrderDirection) -> String {
+    format!(
+        r"
+    SELECT
+        metadatas.address,
+        metadatas.name,
+        metadatas.seller_fee_basis_points,
+        metadatas.update_authority_address,
+        metadatas.mint_address,
+        metadatas.primary_sale_happened,
+        metadatas.uri,
+        metadatas.slot,
+        metadata_jsons.description,
+        metadata_jsons.image,
+        metadata_jsons.category,
+        metadata_jsons.model
+    FROM metadata_jsons
+    INNER JOIN metadatas ON (metadatas.address = metadata_jsons.metadata_address)
+    INNER JOIN (
+        SELECT metadata_collection_keys.collection_address AS collection, MIN(listings.price) * COUNT(listings.price) AS market_cap
+            FROM listings
+            INNER JOIN metadatas on (listings.metadata = metadatas.address)
+            INNER JOIN metadata_collection_keys on (metadatas.address = metadata_collection_keys.metadata_address)
+            WHERE
+                ($1 IS NULL OR metadata_collection_keys.collection_address = ANY($1))
+                AND listings.purchase_id IS NULL
+                AND listings.canceled_at IS NULL
+            GROUP BY metadata_collection_keys.collection_address
+            ORDER BY market_cap {order_direction}
+            LIMIT $2
+            OFFSET $3
```

Comment:
Same question as before on doing the price market_cap in outer query then joining in the results table int the one returning nfts.

Reply:
<img width="1114" alt="Screen Shot 2022-06-30 at 2 48 24 PM" src="https://user-images.githubusercontent.com/2388118/176680776-3915d165-006c-49a0-8399-54e84e645032.png">

There is a long seq scan on `listings.purchase_id IS NULL AND listings.canceled_at IS NULL` that is a pretty common lookup path so good candidate for a joint index.

Reply:
> <img alt="Screen Shot 2022-06-30 at 2 48 24 PM" width="1114" src="https://user-images.githubusercontent.com/2388118/176680776-3915d165-006c-49a0-8399-54e84e645032.png">
> 
> There is a long seq scan on `listings.purchase_id IS NULL AND listings.canceled_at IS NULL` that is a pretty common lookup path so good candidate for a joint index.



> <img alt="Screen Shot 2022-06-30 at 2 48 24 PM" width="1114" src="https://user-images.githubusercontent.com/2388118/176680776-3915d165-006c-49a0-8399-54e84e645032.png">
> 
> There is a long seq scan on `listings.purchase_id IS NULL AND listings.canceled_at IS NULL` that is a pretty common lookup path so good candidate for a joint index.

This went a little over my SQL-noob head. Are you saying we should create another table/index for active listings, or that I should be doing something additional in my query?

Reply:
Based on feedback of the market cap calculation, I am omitting these conditions, so not creating the index for this PR.